@petejohanson, the `python3-dev` switcharoo was to bypass this error ...

```
#5 30.67   x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DPSUTIL_POSIX=1 -DPSUTIL_SIZEOF_PID_T=4 -DPSUTIL_VERSION=580 -DPSUTIL_LINUX=1 -I/usr/include/python3.7m -c psutil/_psutil_common.c -o build/temp.linux-x86_64-3.7/psutil/_psutil_common.o
#5 30.67   psutil/_psutil_common.c:9:10: fatal error: Python.h: No such file or directory
#5 30.67    #include <Python.h>
#5 30.67             ^~~~~~~~~~
#5 30.67   compilation terminated.
#5 30.67   error: command 'x86_64-linux-gnu-gcc' failed with exit status 1
```
... as part of the pip installation ...
```
#5 [common 2/2] RUN   apt-get -y update   && apt-get -y install --no-install-recommends   ccache   file   gcc   gcc-multilib   git   gperf   make   ninja-build   python3   python3-pip   python3-setuptools   python3-wheel   && echo deb http://deb.debian.org/debian buster-backports main >> /etc/apt/sources.list   && apt-get -y update   && apt-get -y -t buster-backports install --no-install-recommends   cmake   && pip3 install   -r https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/v2.5.0/scripts/requirements-base.txt   && apt-get remove -y --purge   python3-pip   python3-setuptools   python3-wheel   && apt-get clean   && rm -rf /var/lib/apt/lists/*
```

I'm not a Python expert, so it was an educated guess.

According to https://packages.debian.org/sid/python3-dev, it's:
> Header files, a static library and development tools for building Python modules, extending the Python interpreter or embedding Python in applications. 

In your opinion, should `python3-dev` be removed (and the end of that step) or left in the `build` stage please?

--- a/.github/workflows/containers.yml
+++ b/.github/workflows/containers.yml
@@ -1,7 +1,7 @@
 name: Containers
 
 env:
-  zephyr-version: 2.4.0
+  zephyr-version: 2.5.0
   zephyr-sdk-version: 0.12.4
   sha-abbrev-length: 12
   no-cache: ${{ secrets.NO_CACHE != null || github.event_name == 'workflow_dispatch' || github.event_name == 'schedule' }}

--- a/Dockerfile
+++ b/Dockerfile
@@ -18,6 +18,7 @@ RUN \
   make \
   ninja-build \
   python3 \
+  python3-dev \
   python3-pip \
   python3-setuptools \
   python3-wheel \
@@ -28,6 +29,7 @@ RUN \
   && pip3 install \
   -r https://raw.githubusercontent.com/zephyrproject-rtos/zephyr/v${ZEPHYR_VERSION}/scripts/requirements-base.txt \
   && apt-get remove -y --purge \
+  python3-dev \
   python3-pip \
   python3-setuptools \
   python3-wheel \

